cocos/ui/UILayoutComponent.cpp

```diff
@@ -667,7 +667,23 @@ namespace ui {
         _owner->setPosition(ownerPosition);
         _owner->setContentSize(ownerSize);
 
-        ui::Helper::doLayout(_owner);
+        if (typeid(*_owner) == typeid(PageView))
+        {
+            PageView* page = static_cast<PageView*>(_owner);
+            page->forceDoLayout();
+
+            Vector<Layout*> _layoutVector = page->getPages();
```

Comment:
I suggest to use `auto&` and `range for loop here`, it's modern c++ style and it's more clean and compact.